src/main/java/org/apache/parquet/hadoop/BdecParquetWriter.java

```diff
@@ -0,0 +1,271 @@
+/*
+ * Copyright (c) 2022 Snowflake Computing Inc. All rights reserved.
+ */
+
+package org.apache.parquet.hadoop;
+
+import static net.snowflake.ingest.utils.Constants.MAX_CHUNK_SIZE_IN_BYTES;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import net.snowflake.ingest.utils.Constants;
+import net.snowflake.ingest.utils.ErrorCode;
+import net.snowflake.ingest.utils.SFException;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.parquet.column.ColumnDescriptor;
+import org.apache.parquet.column.ParquetProperties;
+import org.apache.parquet.column.values.factory.DefaultV1ValuesWriterFactory;
+import org.apache.parquet.crypto.FileEncryptionProperties;
+import org.apache.parquet.hadoop.api.WriteSupport;
+import org.apache.parquet.hadoop.metadata.CompressionCodecName;
+import org.apache.parquet.io.DelegatingPositionOutputStream;
+import org.apache.parquet.io.OutputFile;
+import org.apache.parquet.io.ParquetEncodingException;
+import org.apache.parquet.io.PositionOutputStream;
+import org.apache.parquet.io.api.Binary;
+import org.apache.parquet.io.api.RecordConsumer;
+import org.apache.parquet.schema.MessageType;
+import org.apache.parquet.schema.PrimitiveType;
+
+/**
+ * BDEC specific parquet writer.
+ *
+ * <p>Resides in parquet package because, it uses {@link InternalParquetRecordWriter} and {@link
+ * CodecFactory} that are package private.
+ */
+public class BdecParquetWriter implements AutoCloseable {
+  private final InternalParquetRecordWriter<List<Object>> writer;
+  private final CodecFactory codecFactory;
+
+  public BdecParquetWriter(
+      ByteArrayOutputStream stream,
+      MessageType schema,
+      Map<String, String> extraMetaData,
+      String channelName)
+      throws IOException {
+    OutputFile file = new ByteArrayOutputFile(stream);
+    ParquetProperties encodingProps = createParquetProperties();
+    Configuration conf = new Configuration();
+    WriteSupport<List<Object>> writeSupport =
+        new BdecWriteSupport(schema, extraMetaData, channelName);
+    WriteSupport.WriteContext writeContext = writeSupport.init(conf);
+
+    ParquetFileWriter fileWriter =
+        new ParquetFileWriter(
+            file,
+            schema,
+            ParquetFileWriter.Mode.CREATE,
+            ParquetWriter.DEFAULT_BLOCK_SIZE,
+            ParquetWriter.MAX_PADDING_SIZE_DEFAULT,
+            encodingProps.getColumnIndexTruncateLength(),
+            encodingProps.getStatisticsTruncateLength(),
+            encodingProps.getPageWriteChecksumEnabled(),
+            (FileEncryptionProperties) null);
+    fileWriter.start();
+
+    codecFactory = new CodecFactory(conf, ParquetWriter.DEFAULT_PAGE_SIZE);
```

Comment:
Could you comment on this why we set `ParquetWriter.DEFAULT_PAGE_SIZE` as discussed offlien?

Reply:
done